src/Features/Core/Portable/PullMemberUp/AbstractMembersPullerService.cs

```diff
@@ -299,9 +310,66 @@ private static ISymbol GetSymbolsToPullUp(MemberAnalysisResult analysisResult)
             }
 
             destinationEditor.ReplaceNode(destinationSyntaxNode, (syntaxNode, generator) => newDestination);
+
+            // add imports by combining source and destination imports, then taking out unneccessary
+            // imports that we don't need from the destination. If source and destination are the same document
+            // or have the same imports, this doesn't do anything
+            var destinationDocument = destinationEditor.GetChangedDocument();
+            var destinationRoot = await destinationDocument.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var sourceImports = await GetImportsAsync(document, cancellationToken).ConfigureAwait(false);
+            var destinationImports = await GetImportsAsync(destinationDocument, cancellationToken).ConfigureAwait(false);
+            var destinationTrivia = GetLeadingTriviaBeforeFirstMember(destinationRoot);
+            var addImportsService = destinationDocument.Project.LanguageServices.GetRequiredService<IAddImportsService>();
+            var semanticModel = await destinationDocument.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            destinationDocument = destinationDocument.WithSyntaxRoot(addImportsService.AddImports(
+                semanticModel.Compilation,
+                destinationRoot,
+                null,
+                sourceImports,
+                destinationEditor.Generator,
+                options.PlaceSystemNamespaceFirst,
+                document.CanAddImportsInHiddenRegions(),
+                cancellationToken));
+
+            var removeImportsService = destinationDocument.Project.LanguageServices.GetRequiredService<IRemoveUnnecessaryImportsService>();
+            destinationDocument = await removeImportsService.RemoveUnnecessaryImportsAsync(
+                destinationDocument,
+                node => !destinationImports.Any(destinationNode => IsValidUnnecessaryImport((TUsingOrAliasSyntax)node, destinationNode)),
```

Comment:
you should pass in a predicate that only removes any of hte usings that *you* added.  you can tell this by adding an annotation to yoru own usings, and then looking for that annotation.  this addresses teh case (which you should test for) where the destination has an unused usings that matches a using you are considering moving over. in that case you do *not* want to remove the using from teh destination as it was already there.  you only want to remove things you actually added.